jaeger-core/src/main/java/com/uber/jaeger/metrics/Metrics.java

```diff
@@ -32,7 +32,7 @@ public Metrics(StatsFactory factory) {
         continue;
       }
 
-      StringBuilder metricBuilder = new StringBuilder("jaeger.");
+      StringBuilder metricBuilder = new StringBuilder("jaeger:");
```

Comment:
I have a question, 
go-client starts with "metric:",  don't we need to be consistent with go-client?

Reply:
Yes, the same metrics reported via go-client should have the same name as the java-client. I actually compared the metric names + the output from here: https://github.com/jaegertracing/jaeger/issues/572 . Do you mean that metrics aren't shown like this:

```
# HELP jaeger:finished_spans jaeger:finished_spans
# TYPE jaeger:finished_spans counter
jaeger:finished_spans 0
# HELP jaeger:reporter_queue_length jaeger:reporter_queue_length
# TYPE jaeger:reporter_queue_length gauge
jaeger:reporter_queue_length 0
# HELP jaeger:reporter_spans jaeger:reporter_spans
# TYPE jaeger:reporter_spans counter
jaeger:reporter_spans{result="dropped"} 0
jaeger:reporter_spans{result="err"} 0
jaeger:reporter_spans{result="ok"} 0
```

Reply:
I think you're right. 
From go-client,  "metric:" is used for extracting tags from struct Metrics.
"jaeger" is used for metric name. :)

Reply:
I realize this PR was to unify with the Go client. However by departing from the dot notation with micrometer you lose the hierarchical nature for config.  That way users can `management.metrics.enable.jaegar=false` if they want to disable/customize those metrics.

I also recommend using dots in the metric names like `@Metric(name = "finished.spans")`  The end result to Prometheus will replace the dots with underscores (or format the metric to match the backend).

Reply:
> That way users can management.metrics.enable.jaegar=false if they want to disable/customize those metrics.

@checketts could you please elaborate on this? Where would you use this ^ dot-delimited syntax?

Reply:
For Spring Boot applications you can set those property to control which metrics you want to enable/disable. Similar to enabling/disabling logging levels.

For non-Spring applications then the user would add in their own MeterFilter like so `myRegistry.config().filter(MeterFilter.denyNameStartsWith("jaegar.reporter"))`. In that case the colon wouldn't break the usage, but it does deviate from the expected notation.

Reply:
mentioned this in https://github.com/jaegertracing/jaeger/issues/572#issuecomment-372483488

what if everything was separated with underscore?

Reply:
Only dots imply the hierarchical naming for Micrometer config. If using a colon or underscore, then there is no tokenization and each metric name is treated as a single value.  

So I believe the Spring property would be: `management.metrics.enable.jaeger:reporter_spans: false` but yaml will complain with colon's in the key so it would need to be escaped.

If the colon was replaced then it will avoid the yaml problems. But it will still not be tokenized so Spring Users would have to disable each metric individually.

Just to be clear: A metric named `jaeger.finished.spans` in Micrometer will be output as `jaeger_finished_spans` in Prometheus. So it shouldn't impact that aspect of #572

Reply:
@checketts what is the purpose of enabling/disabling metrics via config file?

Reply:
Also you said 

> I also recommend using dots in the metric names like `@Metric(name = "finished.spans")`

Finished spans is a single metric, why do you gain by treating that name as hierarchical?

fwiw I am trying to reconcile conflicting asks (see https://github.com/jaegertracing/jaeger/issues/732)

Reply:
The purpose of enabling/disabling metrics via config file is similar to enabling/disabling debug logging. The user of client can disable metrics they don't need (or want to pay for when it comes to 3rd party monitoring systems like DataDog or CloudWatch).  

The dot name separation supports hierarchy, since to Micrometer's view the metric's complete name is `jaeger.finished.spans`, so a user would be able to enable/disable all the Jaeger metrics easily. Specifically regarding the `finished.spans` section, Micrometer formats the metric name according to the corresponding backend (underscores for Prometheus, camel case for Atlas, etc.)  

I don't think the dot notion will affect #732 much since Micrometer writes out the metrics, trying to match the corresponding backend. So the underscores will be preserved on output to Prometheus.

The colon is really the only sticking point. For users than really want the colon, they could add a `MeterFilter` like so:

```
registry.config().filter(new MeterFilter() {
  public Meter.Id map(Meter.Id id) {
    if(id.getName().startsWith("jaegar.")) {
      return new Meter.Id(id.getName().replace("jaegar.", "jaegar:"), allTags, id.getBaseUnit(), id.getDescription(), id.getType());
    }
    
    return id;
  }
});
```

You could even include that `MeterFilter` in your client and let the user wire it up if they need it.  

(As a  I'm also not a fan of the colon since it is typically used in [recording rules](https://prometheus.io/docs/practices/rules/#examples) in Prometheus and implies an aggregation has been applied)